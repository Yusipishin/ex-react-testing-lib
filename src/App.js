import {useEffect, useState} from "react";

function App() {
    const [data, setData] = useState(null);
    const [toggle, setToggle] = useState(false)

    const onClick = () => setToggle(prev => !prev)

    useEffect(() => {
        setTimeout(() => {
            setData({})
        }, 100)
    }, []);

    return (
    <div>
        {toggle && <div data-testid="toggle-elem" >toggle</div>}
        {data && <div style={{color: 'red'}}>DATA</div>}
        <h1>Hello World!</h1>
        <button data-testid="toggle-btn" onClick={onClick}>Click me!</button>
        <input type="text" placeholder="input vfalue..."/>
    </div>
)
}

export default App;
