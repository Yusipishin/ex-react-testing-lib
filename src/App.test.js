import { render, screen } from '@testing-library/react';
import App from './App';

describe('APP TEST', () => {
    test('renders learn react link', () => {
      render(<App />);

      const helloWorldElement = screen.getByText(/hello world!/i)
      const btn = screen.getByRole('button')
      const input = screen.getByPlaceholderText(/input vfalue.../i)

      expect(helloWorldElement).toBeInTheDocument()
      expect(btn).toBeInTheDocument()
      expect(input).toMatchSnapshot()
  });

    // find -- для асинхронного кода
    // get -- 100% получение элемента
    // query -- доказать отсутствие

    test('renders learn react link', async() => {
      render(<App />);

      // const helloWorldElement = screen.queryByText(/hello142/i)
      // expect(helloWorldElement).toBeNull()

      const helloWorldElement = await screen.findByText(/data/i)
      expect(helloWorldElement).toBeInTheDocument()
      expect(helloWorldElement).toHaveStyle({color: 'red'})

  });
});
