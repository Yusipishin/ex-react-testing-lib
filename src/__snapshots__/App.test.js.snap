// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP TEST renders learn react link 1`] = `
<input
  placeholder="input vfalue..."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
